### DISCUSSION TITLE
Types are assignable to type 'ComponentType | ExoticComponent<{}>

### DISCUSSION THREAD

**Discussion by likhitabe** [TroyAlford/react-jsx-parser#259] [Q&A] [answered]
Hi Everyone, I'm trying to use this dependency for my project by passing below props and components prop with my custom functional components as shown below.
`const jsxParser = new JsxParser({ jsx: JSXContent, bindings:  { data , callbacks, page } ,components:{Input, Header} });`
But when I'm trying to run the code, its working fine but i'm getting type errors in build 

`Type '({ name, data, pattern, placeholder, style, maxLength, minLength, callbacks, required, left, top, height, width, ...props }: InputProps) => JSX.Element' is not assignable to type 'ComponentType | ExoticComponent<{}>'.
  Type '({ name, data, pattern, placeholder, style, maxLength, minLength, callbacks, required, left, top, height, width, ...props }: InputProps) => JSX.Element' is not assignable to type 'FunctionComponent<{}>'.
`
can someone help me on this ?

**Comment by TroyAlford** [comment, marked as answer]
Your typing is too narrow. I believe you want `React.ComponentType<TProps>` in order to handle class components or functional components
